### ISSUE TITLE
Run CI for micropipenv

### ISSUE THREAD

**Issue by fridex** [thoth-station/aicoe-ci#27]
**Is your feature request related to a problem? Please describe.**

It would be great if we could automate testing of micropipenv for each PR. As micropipenv is on a different layer, it might be tricky to run the CI with the same configuration as for the other components of Thoth.

**Describe the solution you'd like**

Execute micropipenv testsuite using ``tox``. We also need Python interpreters in different versions to verify the correctness.  See https://github.com/thoth-station/micropipenv/pull/72 for more info.

**Describe alternatives you've considered**

As micropipenv is not tightly bound to Thoth itself, we could use centos CI or travis to execute the test suite. As the requirements are pretty straightforward we could let these solutions run the CI not to maintain big changes on our side.

**Comment by fridex** [comment]
CC @harshad16 @frenzymadness @goern

**Comment by frenzymadness** [comment]
I have this [container image](https://github.com/fedora-python/fedora-python-tox) ready with all dependencies micropipenv needs for testing and I know how to configure it in Travis. So, if the Travis is a way to go, I can prepare something. I don't know much about Centos CI.

Recently, @encukou has prepared [tox github action](https://github.com/marketplace/actions/python-tox-on-fedora) with mentioned container so we might try that as well. But I have zero experience with GH actions.

**Comment by harshad16** [comment]
we can extend the pipeline to run the tox test with the [container image](https://github.com/fedora-python/fedora-python-tox) for micropipenv. I will include a sample to the existing pipeline, and we can check the compatibility with it. @fridex

**Comment by frenzymadness** [comment]
> we can extend the pipeline to run the tox test with the [container image](https://github.com/fedora-python/fedora-python-tox) for micropipenv.

What pipeline do you mean?

**Comment by frenzymadness** [comment]
I've tried three possibilities here:

1. Pure Travis [config](https://github.com/frenzymadness/micropipenv/commit/e1bdfb20745ff875a91eb25b93bbf6ac53406c78) & [result](https://travis-ci.org/github/frenzymadness/micropipenv/builds/693722229) — Simple configuration and the tests run on Ubuntu which should make no difference because we depends only on Python packages from PyPI.
2. Travis with Fedora container [config](https://github.com/frenzymadness/micropipenv/commit/198c3de79860dd54c4c896692b07aaf05d0f9aaf) & [result](https://travis-ci.org/github/frenzymadness/micropipenv/builds/693724476) — Closer to our target environment and new Pythons are available in Fedora sooner than in Travis.
3. GH actions with Fedora container [testing PR](https://github.com/frenzymadness/micropipenv/pull/1) — Looks good to me as well, has nice integration to GH but the action itself is still in pre-release (but maintained by fedora-python).

**Comment by harshad16** [comment]
> > we can extend the pipeline to run the tox test with the [container image](https://github.com/fedora-python/fedora-python-tox) for micropipenv.
> 
> What pipeline do you mean?

we developed aicoe-ci based on openshift-pipeline/tekton-pipeline, it runs checks as task(similar to a job). Earlier as i stated we can embed another task for tox test, based upon the container image earlier mentioned in the thread.

**Comment by frenzymadness** [comment]
So, which one we want to use?

**Comment by frenzymadness** [comment]
Travis with Fedora container implemented in https://github.com/thoth-station/micropipenv/pull/87

This issue can be closed.

**Comment by fridex** [comment]
Let's close this, micropipenv is tested with Travis CI now. If there is a plan to add support for Thoth CI, we can still do it besides Travis as discussed on the call.

Thanks @frenzymadness @harshad16 :+1:

**Comment by harshad16** [comment]
Ack, tox test is already included in the pipeline. However, it needs to be refined more, it would run in parallel with the Travis test. It wouldn't cause any issue in the pull request.
